compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1960,12 +1963,15 @@ class Typer extends Namer
     object splitter extends tpd.TreeMap {
       val patBuf = new mutable.ListBuffer[Tree]
       override def transform(tree: Tree)(implicit ctx: Context) = tree match {
-        case Typed(Splice(pat), tpt) =>
+        case Typed(Splice(pat), tpt) if !tpt.tpe.derivesFrom(defn.RepeatedParamClass) =>
           val exprTpt = AppliedTypeTree(TypeTree(defn.QuotedExprType), tpt :: Nil)
           transform(Splice(Typed(pat, exprTpt)))
         case Splice(pat) =>
           try patternHole(tree)
-          finally patBuf += pat
+          finally {
+            val pat1 = pat.subst(defn.RepeatedParamClass :: Nil, defn.SeqClass :: Nil)
+            patBuf += pat1
+          }
```

Comment:
The change here and above looks dubious to me. @odersky could you have a look?

Reply:
I am not sure a global substitution is the right thing here. I fear it would also affect nested varargs methods that would then become Seq methods. I'd do instead:
```scala
val patType = pat.tpe.widen
val patType1 = patType.underlyingIfRepeated(isJava = false)
val pat1 = if (patType eq patType1) pat else pat.withType(patType1)
```

Reply:
I use this code now and also updated another place that had the `subst` to use `underlyingIfRepeated`.